tests/test_trial.py

```diff
@@ -764,9 +765,325 @@ def test_study_id():
     assert trial._study_id == trial.study._study_id
 
 
-def test_frozen_trial_validate():
+def test_frozen_trial_eq_ne():
+    # type: () -> None
+
+    trial = _create_frozen_trial()
+
+    trial_other = copy.copy(trial)
+    assert trial == trial_other
+
+    trial_other.value = 0.3
+    assert trial != trial_other
+
+
+def test_frozen_trial_lt():
     # type: () -> None
 
+    trial = _create_frozen_trial()
+
+    trial_other = copy.copy(trial)
+    assert not trial < trial_other
+
+    trial_other.number = trial.number + 1
+    assert trial < trial_other
+    assert not trial_other < trial
+
+    with pytest.raises(TypeError):
+        trial < 1
+
+    assert trial <= trial_other
+    assert not trial_other <= trial
+
+    with pytest.raises(TypeError):
+        trial <= 1
+
+    # A list of FrozenTrials is sortable.
+    trials = [trial_other, trial]
+    trials.sort()
+    assert trials[0] is trial
+    assert trials[1] is trial_other
+
+
+def _create_frozen_trial():
+    # type: () -> FrozenTrial
+
+    return FrozenTrial(
+        number=0,
+        trial_id=0,
+        state=TrialState.COMPLETE,
+        value=0.2,
+        datetime_start=datetime.datetime.now(),
+        datetime_complete=datetime.datetime.now(),
+        params={"x": 10},
+        distributions={"x": UniformDistribution(5, 12)},
+        user_attrs={},
+        system_attrs={},
+        intermediate_values={},
+    )
+
+
+def test_frozen_trial_repr():
+    # type: () -> None
+
+    trial = FrozenTrial(
+        number=0,
+        trial_id=0,
+        state=TrialState.COMPLETE,
+        value=0.2,
+        datetime_start=datetime.datetime.now(),
+        datetime_complete=datetime.datetime.now(),
+        params={"x": 10},
+        distributions={"x": UniformDistribution(5, 12)},
+        user_attrs={},
+        system_attrs={},
+        intermediate_values={},
+    )
+
+    assert trial == eval(repr(trial))
+
+
+@parametrize_storage
+def test_frozen_trial_sampling(storage_init_func):
+    # type: (Callable[[], storages.BaseStorage]) -> None
+
+    def objective(trial: BaseTrial) -> float:
+
+        trial.suggest_uniform("a", 0, 10)
+        trial.suggest_loguniform("b", 0.1, 10)
+        trial.suggest_discrete_uniform("c", 0, 10, 1)
+        trial.suggest_int("d", 0, 10)
+        trial.suggest_categorical("e", ["foo", "bar", "baz"])
+        trial.suggest_int("f", 1, 10, log=True)
+
+        return 1.0
```

Comment:
It is more informative to make the objective dependent on the suggested params.
```suggestion
    def objective(trial: BaseTrial) -> float:

        a = trial.suggest_uniform("a", 0, 10)
        b = trial.suggest_loguniform("b", 0.1, 10)
        c = trial.suggest_discrete_uniform("c", 0, 10, 1)
        d = trial.suggest_int("d", 0, 10)
        e = trial.suggest_categorical("e", [0, 1, 2])
        f = trial.suggest_int("f", 1, 10, log=True)

        return a + b + c + d + e + f
```

Reply:
Definitely! Thanks!